### PR TITLE
[PWA] WebAppManifest: screenshots and shortcuts support

### PR DESCRIPTION
**Pull request by pizzi80**
- WebAppManifest Screenshots and Shortcuts
- "final" on inner classes effective final fields
- typo

I tried to follow the documentation standard (which is very informative btw :+1: )

### CODE REVIEW

File: src/main/java/org/omnifaces/resourcehandler/WebAppManifest.java [hunk 1]
```diff
@@ -635,4 +654,106 @@ public int hashCode() {
...
+     * To be used in {@link WebAppManifest#getScreenshots()}
+     */
+    protected static final class Screenshot {
+
+        private final ImageResource image;
+        private final ScreenshotFormFactor formFactor;
+
+        public String getSrc() { return image.getSrc(); }
+        public String getSizes() { return image.getSizes(); }
+        public String getType() { return image.getType(); }
```
Comment (BalusC) [inline src/main/java/org/omnifaces/resourcehandler/WebAppManifest.java]: Please do not deviate from code format seen in existing source code.

File: src/main/java/org/omnifaces/resourcehandler/WebAppManifest.java [hunk 2]
```diff
@@ -635,4 +654,106 @@ public int hashCode() {
         }
     }
 
+    /**
+     * A {@link Screenshot} form factor
+     * @see <a href="https://developer.mozilla.org/en-US/docs/Web/Manifest/screenshots#form_factor">https://developer.mozilla.org/en-US/docs/Web/Manifest/screenshots#form_factor</a>
+     */
+    protected enum ScreenshotFormFactor { narrow,wide; }
```
Comment (BalusC) [inline src/main/java/org/omnifaces/resourcehandler/WebAppManifest.java]: Please do not deviate from code format seen in existing source code.

File: src/main/java/org/omnifaces/resourcehandler/WebAppManifest.java [hunk 3]
```diff
@@ -635,4 +654,106 @@ public int hashCode() {
         }
     }
 
+    /**
+     * A {@link Screenshot} form factor
+     * @see <a href="https://developer.mozilla.org/en-US/docs/Web/Manifest/screenshots#form_factor">https://developer.mozilla.org/en-US/docs/Web/Manifest/screenshots#form_factor</a>
```
Comment (BalusC) [inline src/main/java/org/omnifaces/resourcehandler/WebAppManifest.java:659]: Missing `@version`.

File: src/main/java/org/omnifaces/resourcehandler/WebAppManifest.java [hunk 4]
```diff
@@ -635,4 +654,106 @@ public int hashCode() {
...
+     */
+    protected enum ScreenshotFormFactor { narrow,wide; }
+
+    /**
+     * To be used in {@link WebAppManifest#getScreenshots()}
```
Comment (BalusC) [inline src/main/java/org/omnifaces/resourcehandler/WebAppManifest.java:679]: Missing `@version`.

File: src/main/java/org/omnifaces/resourcehandler/WebAppManifest.java [hunk 5]
```diff
@@ -635,4 +654,106 @@ public int hashCode() {
...
+        public int hashCode() {
+            return Objects.hash(image, formFactor);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (!(o instanceof Screenshot)) return false;
+            Screenshot screenshot = (Screenshot) o;
+            return Objects.equals(image, screenshot.image) && Objects.equals(formFactor, screenshot.formFactor);
```
Comment (BalusC) [inline src/main/java/org/omnifaces/resourcehandler/WebAppManifest.java]: Please do not deviate from code format seen in existing source code.

File: src/main/java/org/omnifaces/resourcehandler/WebAppManifest.java [hunk 6]
```diff
@@ -635,4 +654,106 @@ public int hashCode() {
...
+        }
+    }
+
+    /**
+     * to be used {@link WebAppManifest#getShortcuts()}
```
Comment (BalusC) [inline src/main/java/org/omnifaces/resourcehandler/WebAppManifest.java:743]: Missing `@version`.

File: src/main/java/org/omnifaces/resourcehandler/WebAppManifest.java [hunk 7]
```diff
@@ -635,4 +654,106 @@ public int hashCode() {
...
+        public int hashCode() {
+            return Objects.hash(url);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (!(o instanceof Shortcut)) return false;
+            Shortcut shortcut = (Shortcut) o;
+            return Objects.equals(url, shortcut.url);
```
Comment (BalusC) [inline src/main/java/org/omnifaces/resourcehandler/WebAppManifest.java:805]: Please do not deviate from code format seen in existing source code.

**Comment by BalusC** [comment]
There are test failures
```
Error:  Failures: 
Error:    PWAResourceHandlerIT.verifyManifest:67 expected: <{"categories":[],"dir":"auto","display":"browser","icons":[{"sizes":"512x512","src":"\/PWAResourceHandlerIT\/jakarta.faces.resource\/icon.png.xhtml?v=1","type":"image\/png"}],"lang":"en","name":"PWAResourceHandlerIT","orientation":"any","prefer_related_applications":false,"related_applications":[],"start_url":"http:\/\/127.0.0.1:8080\/PWAResourceHandlerIT\/"}> but was: <{"categories":[],"dir":"auto","display":"browser","icons":[{"sizes":"512x512","src":"\/PWAResourceHandlerIT\/jakarta.faces.resource\/icon.png.xhtml?v=1","type":"image\/png"}],"lang":"en","name":"PWAResourceHandlerIT","orientation":"any","prefer_related_applications":false,"related_applications":[],"screenshots":[],"shortcuts":[],"start_url":"http:\/\/127.0.0.1:8080\/PWAResourceHandlerIT\/"}>
[INFO] 
Error:  Tests run: 143, Failures: 1, Errors: 0, Skipped: 2
```

**Comment by pizzi80** [comment]
should be fixed ...

**Pull request merged**